src/charm.py

```diff
@@ -522,6 +536,22 @@ def _on_admin_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
         self._handle_status_update_config(event)
         self._update_kratos_endpoints_relation_data(event)
 
+    def _apply_network_policies(self, event: HookEvent) -> None:
+        if not self.unit.is_leader():
+            return
+
+        try:
+            self.network_policy_handler.apply_ingress_policy(
+                [
+                    (KRATOS_PUBLIC_PORT, [self.public_ingress.relation]),
+                    (KRATOS_ADMIN_PORT, [self.admin_ingress.relation]),
+                    (38812, []),
+                    (38813, []),
+                ]
+            )
+        except NetworkPoliciesHandlerError:
+            event.defer()
```

Comment:
Shouldn't we simply change the status to blocked (or not catch the error and let the charm go to error state)? If something is broken we may not want to be deferring forever